transactions/sell_item.cdc

```diff
@@ -1,43 +1,75 @@
+    let storefront: &NFTStorefrontV2.Storefront
+    var saleCuts: [NFTStorefrontV2.SaleCut]
+    var marketplacesCap: [Capability<&AnyResource{FungibleToken.Receiver}>]
 
     prepare(acct: AuthAccount) {
+        self.saleCuts = []
+        self.marketplacesCap = []
         // We need a provider capability, but one is not provided by default so we create one if needed.
         let exampleNFTCollectionProviderPrivatePath = /private/exampleNFTCollectionProviderForNFTStorefront
 
-        self.flowReceiver = acct.getCapability<&FlowToken.Vault{FungibleToken.Receiver}>(/public/flowTokenReceiver)!
+        self.flowReceiver = acct.getCapability<&{FungibleToken.Receiver}>(/public/flowTokenReceiver)!
         assert(self.flowReceiver.borrow() != nil, message: "Missing or mis-typed FlowToken receiver")
 
-        if !acct.getCapability<&ExampleNFT.Collection{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath)!.check() {
-            acct.link<&ExampleNFT.Collection{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath, target: /storage/NFTCollection)
+        if !acct.getCapability<&{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath)!.check() {
+            acct.link<&{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath, target: ExampleNFT.CollectionStoragePath)
         }
 
-        self.exampleNFTProvider = acct.getCapability<&ExampleNFT.Collection{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath)!
+        self.exampleNFTProvider = acct.getCapability<&{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>(exampleNFTCollectionProviderPrivatePath)!
+        let collection = acct
+            .getCapability(ExampleNFT.CollectionPublicPath)
+            .borrow<&{ExampleNFT.ExampleNFTCollectionPublic}>()
+            ?? panic("Could not borrow a reference to the collection")
+        var totalRoyaltyCut = 0.0
+        let effectiveSaleItemPrice = saleItemPrice - commissionAmount
+        let nft = collection.borrowExampleNFT(id: saleItemID)!
+        // Check whether the NFT implements the MetadataResolver or not.
+        if nft.getViews().contains(Type<MetadataViews.Royalties>()) {
+            let royaltiesRef = nft.resolveView(Type<MetadataViews.Royalties>())?? panic("Unable to retrieve the royalties")
+            let royalties = (royaltiesRef as! MetadataViews.Royalties).getRoyalties()
+            for royalty in royalties {
+                // TODO - Verify the type of the vault and it should exists
+                self.saleCuts.append(NFTStorefrontV2.SaleCut(receiver: royalty.receiver, amount: royalty.cut * effectiveSaleItemPrice))
+                totalRoyaltyCut = totalRoyaltyCut + royalty.cut * effectiveSaleItemPrice
```

Comment:
```suggestion
                totalRoyaltyCut = totalRoyaltyCut + (royalty.cut * effectiveSaleItemPrice)
```

not sure how math works in cadence, but probably needs to be like this ☝️ 

or maybe re-arrange like

```
                totalRoyaltyCut = royalty.cut * effectiveSaleItemPrice + totalRoyaltyCut
```